server/src/test/scala/com/ossum/riddl/lsp/server/RiddlLSPTextDocumentSpec.scala

```diff
@@ -160,12 +153,42 @@ class RiddlLSPTextDocumentSpec
 
     "successfully change empty.riddl" in new ChangeEmptyFileSpec {}
 
-    "successfully save empty.riddl" in new ChangeEmptyFileSpec {
+     */
+
+    "successfully save empty.riddl" in new ChangeEmptyFileSpec
+      with CompletionRequestSpec
+      with DiagnosticRequestSpec {
+
+      // ensuring state before the test
+      position.setLine(1)
+      position.setCharacter(1)
+      requestCompletion()
+      completionResultF.asScala.failed.futureValue.getMessage mustEqual "Document is empty"
+
+      // Need to actually change the file before running the test
+      var p = new java.io.PrintWriter(new File(emptyDocURI))
+      try { p.println(textChange) }
+      finally { p.close() }
+
       val saveNotification = new DidSaveTextDocumentParams()
       saveNotification.setTextDocument(textDocumentIdentifier)
       service.didSave(saveNotification)
+
+      position.setLine(1)
+      position.setCharacter(13)
+      requestCompletion()
+
+      whenReady(completionResultF.asScala) { completion =>
+        completion.isRight mustEqual true
+        completion.getRight.getItems.asScala.length mustEqual 1
+        completion.getRight.getItems.asScala.head.getDetail mustEqual
+          """Expected one of ("/*" | "//" | "???" | "application" | "author" | "by" | "command" | "constant" | "context" | "domain" | "epic" | "event" | "graph" | "import" | "include" | "option" | "query" | "record" | "result" | "saga" | "table" | "term" | "type" | "user")"""
+      }
+
+      p = new java.io.PrintWriter(new File(emptyDocURI))
+      try { p.print("") }
+      finally { p.close() }
```

Comment:
I don't understand why this exists. You're doing the thing that you should be checking for? The test case is "successfully save empty.riddl" which is what the above 3 lines of code do.  So, is this test case named right? 
At the very least you also need to query the file system to make sure the file is present and empty. But that's a test of the file system, not a test of your riddl-lsp code.

Reply:
The file has to change to ensure anything testable has happened in the LSP. Otherwise it could be old global data that is causing the change.

I did write the following to you in Slack:
`There is a choice to make here... the request for didSave allows for including the text of the document in case that ends up being more useful than reading the document on the client from the server, assuming it was saved already... so the choice is basically using Source v inserting the doc in the request... this is a field the server requests the client to include

didSave is supposed to be used in a sync call, so it can also be used for providing deltas in case it isn't safe to assume the client has saved or has the only correct version. But I doubt that applies here`

I chose the easiest & most sensible route, which is to assume the editor has saved correctly